src/tools/ceph-dencoder/ceph_dencoder.cc

```diff
@@ -163,6 +164,15 @@ int main(int argc, const char **argv)
 	return 1;
       }
       den->encode(encbl, features | CEPH_FEATURE_RESERVED); // hack for OSDMap
+    } else if (*i == string("add_crc32c")) {
+      if (!encbl.length()) {
+	cerr << "must first encode something" << std::endl;
+	return 1;
+      }
+      auto p = encbl.begin();
+      p += skip;
```

Comment:
Ignore this comment, if it is irrevalent (Not aware of much details of the code),
Should we put assert here, whether `p += skip` is going out of bound or not

Reply:
Yeah, makes sense